### ISSUE TITLE
[Flowbroker] HTTP node execution failed

### ISSUE THREAD

**Issue by betefaber** [dojot/dojot#1650]
The same flow was created in baselines v0.5.0-alpha.3 and v0.4.3-rc.

![fluxo](https://user-images.githubusercontent.com/37310063/90679243-43185180-e236-11ea-9607-df39a03bda6d.png)

The http node was successfully executed on v0.5.0-alpha.3 but failed on v0.4.3-rc.1.

----
**v0.4.3-rc.1: execution failed**
----

```
<18:59:03 19/08/2020> -- DEBUG: Got a message in topic d8e65087-a461-4114-ae7e-1e9e4c442465.
<18:59:03 19/08/2020> -- DEBUG: There are 1 callbacks registered.
<18:59:03 19/08/2020> -- |messenger -- DEBUG: Received message: { value: <Buffer 7b 22 6d 65 74 61 64 61 74 61 22 3a 7b 22 64 65 76 69 63 65 69 64 22 3a 22 63 37 39 35 32 35 22 2c 22 74 65 6e 61 6e 74 22 3a 22 61 64 6d 69 6e 22 2c ... >,
  size: 103,
  key: null,
  topic: 'd8e65087-a461-4114-ae7e-1e9e4c442465',
  offset: 48,
  partition: 0,
  timestamp: 1597863543528 }
<18:59:03 19/08/2020> -- |messenger -- DEBUG: Emitting new event message for subject device-data@admin
<18:59:03 19/08/2020> -- DEBUG: { '3228d834-1434-4b49-9cb4-b24bc44e8e35': [Function] }
<18:59:03 19/08/2020> -- DEBUG: Mapping device c79525 ===into===> event_queue5
<18:59:03 19/08/2020> -- DEBUG: Pre-processing event {"source":"device","message":"{\"metadata\":{\"deviceid\":\"c79525\",\"tenant\":\"admin\",\"timestamp\":1597863543527},\"attrs\":{\"velocidade\":70}}"}
<18:59:03 19/08/2020> -- |dev cache -- DEBUG: retriving data related to admin:c79525 from cache
<18:59:03 19/08/2020> -- DEBUG: [ingestor] got new device event: { metadata: { tenant: 'admin', timestamp: 1597863543527 },
  event: 'publish',
  data:
   { attrs:
      { velocidade: 70,
        carro: 'indefinido',
        device_timeout: '10000',
        protocol: 'mqtt' },
     id: 'c79525' } }
<18:59:03 19/08/2020> -- DEBUG: [executor] will handle node ce40a438.e567c8:switch
<18:59:03 19/08/2020> -- |switch -- DEBUG: Executing switch node...
<18:59:03 19/08/2020> -- |switch -- DEBUG: ... switch node was successfully executed.
<18:59:03 19/08/2020> -- DEBUG: [executor] hop (switch) result: [{"payload":{"velocidade":70,"carro":"indefinido","device_timeout":"10000","protocol":"mqtt"}}]
<18:59:03 19/08/2020> -- DEBUG: Queued event [object Object]
<18:59:03 19/08/2020> -- DEBUG: [executor] will handle node 5d51ecfa.baa4e4:template
<18:59:03 19/08/2020> -- |template -- DEBUG: ... template node was successfully executed.
<18:59:03 19/08/2020> -- DEBUG: [executor] hop (template) result: [{"payload":{"velocidade":70,"carro":"indefinido","device_timeout":"10000","protocol":"mqtt"},"reqBody":"{\"payload\": \"velocidade muito alta: 70 km/h!\"}"}]
<18:59:03 19/08/2020> -- DEBUG: [executor] will handle node 784cd62a.09b088:http
<18:59:03 19/08/2020> -- |http -- DEBUG: Executing http node...
<18:59:03 19/08/2020> -- |http -- DEBUG: HTTP request about to be sent: Url {
  protocol: 'http:',
  slashes: true,
  auth: null,
  host: 'ptsv2.com',
  port: null,
  hostname: 'ptsv2.com',
  hash: null,
  search: null,
  query: null,
  pathname: '/t/3fbhu-1543424220/post',
  path: '/t/3fbhu-1543424220/post',
  href: 'http://ptsv2.com/t/3fbhu-1543424220/post',
  method: 'POST',
  headers: {} }
<18:59:03 19/08/2020> -- WARN: [executor] Node (784cd62a.09b088:http) execution failed. Error: ReferenceError: makeRequest is not defined. Aborting flow d2e07fee-2c96-460a-8a05-dab214b4b5e1 branch execution.
```

----
**v0.5.0-alpha.3: http node was successfully executed**
----

```
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- DEBUG: Got a message in topic admin.device-data.
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- DEBUG: There are 1 callbacks registered.
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- |messenger -- DEBUG: Received message: { value: <Buffer 7b 22 6d 65 74 61 64 61 74 61 22 3a 7b 22 64 65 76 69 63 65 69 64 22 3a 22 64 66 32 34 39 22 2c 22 74 65 6e 61 6e 74 22 3a 22 61 64 6d 69 6e 22 2c 22 ... >,
flowbroker-6dfd89fcff-kwh4p flowbroker   size: 99,
flowbroker-6dfd89fcff-kwh4p flowbroker   key: <Buffer 61 64 6d 69 6e 3a 64 66 32 34 39>,
flowbroker-6dfd89fcff-kwh4p flowbroker   topic: 'admin.device-data',
flowbroker-6dfd89fcff-kwh4p flowbroker   offset: 60,
flowbroker-6dfd89fcff-kwh4p flowbroker   partition: 0,
flowbroker-6dfd89fcff-kwh4p flowbroker   timestamp: 1597861713973 }
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- |messenger -- DEBUG: Emitting new event message for subject device-data@admin
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- DEBUG: { '5cf8fea1-20c8-49d3-adfc-32f69c8b5cd9': [Function] }
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- DEBUG: Pre-processing event {"source":"device","message":"{\"metadata\":{\"deviceid\":\"df249\",\"tenant\":\"admin\",\"timestamp\":1597861713},\"attrs\":{\"velocidade\":70}}"}
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- |dev cache -- DEBUG: retriving data related to admin:df249 from cache
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- DEBUG: [ingestor] got new device event: { metadata: { tenant: 'admin', timestamp: 1597861713 },
flowbroker-6dfd89fcff-kwh4p flowbroker   event: 'publish',
flowbroker-6dfd89fcff-kwh4p flowbroker   data:
flowbroker-6dfd89fcff-kwh4p flowbroker    { attrs:
flowbroker-6dfd89fcff-kwh4p flowbroker       { velocidade: 70,
flowbroker-6dfd89fcff-kwh4p flowbroker         device_timeout: '10000',
flowbroker-6dfd89fcff-kwh4p flowbroker         protocol: 'mqtt',
flowbroker-6dfd89fcff-kwh4p flowbroker         carro: 'indefinido' },
flowbroker-6dfd89fcff-kwh4p flowbroker      id: 'df249' } }
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- DEBUG: Queued event [object Object]
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- DEBUG: [executor] will handle node ce40a438.e567c8:switch
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- |switch -- DEBUG: Executing switch node...
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- |switch -- DEBUG: ... switch node was successfully executed.
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- DEBUG: [executor] hop (switch) result: [{"payload":{"velocidade":70,"device_timeout":"10000","protocol":"mqtt","carro":"indefinido"}}]
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- DEBUG: [executor] will handle node 5d51ecfa.baa4e4:template
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- |template -- DEBUG: ... template node was successfully executed.
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- DEBUG: [executor] hop (template) result: [{"payload":{"velocidade":70,"device_timeout":"10000","protocol":"mqtt","carro":"indefinido"},"reqBody":"{\"payload\": \"velocidade muito alta: 70 km/h!\"}"}]
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- DEBUG: [executor] will handle node 784cd62a.09b088:http
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- |http -- DEBUG: Executing http node...
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:34 19/08/2020> -- |http -- DEBUG: HTTP request about to be sent: Url {
flowbroker-6dfd89fcff-kwh4p flowbroker   protocol: 'http:',
flowbroker-6dfd89fcff-kwh4p flowbroker   slashes: true,
flowbroker-6dfd89fcff-kwh4p flowbroker   auth: null,
flowbroker-6dfd89fcff-kwh4p flowbroker   host: 'ptsv2.com',
flowbroker-6dfd89fcff-kwh4p flowbroker   port: null,
flowbroker-6dfd89fcff-kwh4p flowbroker   hostname: 'ptsv2.com',
flowbroker-6dfd89fcff-kwh4p flowbroker   hash: null,
flowbroker-6dfd89fcff-kwh4p flowbroker   search: null,
flowbroker-6dfd89fcff-kwh4p flowbroker   query: null,
flowbroker-6dfd89fcff-kwh4p flowbroker   pathname: '/t/3fbhu-1543424220/post',
flowbroker-6dfd89fcff-kwh4p flowbroker   path: '/t/3fbhu-1543424220/post',
flowbroker-6dfd89fcff-kwh4p flowbroker   href: 'http://ptsv2.com/t/3fbhu-1543424220/post',
flowbroker-6dfd89fcff-kwh4p flowbroker   method: 'POST',
flowbroker-6dfd89fcff-kwh4p flowbroker   headers: {} }
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:35 19/08/2020> -- |http -- DEBUG: ... http node was successfully executed.
flowbroker-6dfd89fcff-kwh4p flowbroker <18:28:35 19/08/2020> -- DEBUG: [executor] hop (http) result: [{"payload":{"velocidade":70,"device_timeout":"10000","protocol":"mqtt","carro":"indefinido"},"reqBody":"{\"payload\": \"velocidade muito alta: 70 km/h!\"}","responseGet":{"statusCode":200,"headers":{"content-type":"text/plain; charset=utf-8","vary":"Accept-Encoding","access-control-allow-origin":"*","x-cloud-trace-context":"b3ff995743980b7641aa64703a192a16","date":"Wed, 19 Aug 2020 18:28:34 GMT","server":"Google Frontend","content-length":"54","connection":"close"},"responseUrl":"http://ptsv2.com/t/3fbhu-1543424220/post","payload":"Thank you for this dump. I hope you have a lovely day!"}}]
```
![post-v0 5 0-alpha 3](https://user-images.githubusercontent.com/37310063/90687068-a7411280-e242-11ea-8a6f-eeb0ac1b3cf8.png)

Affected baseline: **v0.4.3-rc.1**

**Comment by mprevide** [comment]
#1648

**Comment by betefaber** [comment]
Fixed in v0.4.3-rc.2

- docker

```
flowbroker_1                  | <16:39:24 27/08/2020> -- DEBUG: Got a message in topic 9bf1814d-a93f-45e5-825a-31b85426c0f0.
flowbroker_1                  | <16:39:24 27/08/2020> -- DEBUG: There are 1 callbacks registered.
flowbroker_1                  | <16:39:24 27/08/2020> -- |messenger -- DEBUG: Received message: { value: <Buffer 7b 22 6d 65 74 61 64 61 74 61 22 3a 7b 22 64 65 76 69 63 65 69 64 22 3a 22 65 39 66 39 32 30 22 2c 22 74 65 6e 61 6e 74 22 3a 22 61 64 6d 69 6e 22 2c ... >,
flowbroker_1                  |   size: 103,
flowbroker_1                  |   key: null,
flowbroker_1                  |   topic: '9bf1814d-a93f-45e5-825a-31b85426c0f0',
flowbroker_1                  |   offset: 5,
flowbroker_1                  |   partition: 0,
flowbroker_1                  |   timestamp: 1598546364573 }
flowbroker_1                  | <16:39:24 27/08/2020> -- |messenger -- DEBUG: Emitting new event message for subject device-data@admin
flowbroker_1                  | <16:39:24 27/08/2020> -- DEBUG: { '813daa65-23b5-45b2-83e2-d2cb1a24544c': [Function] }
flowbroker_1                  | <16:39:24 27/08/2020> -- DEBUG: Mapping device e9f920 ===into===> event_queue4
flowbroker_1                  | <16:39:24 27/08/2020> -- DEBUG: Pre-processing event {"source":"device","message":"{\"metadata\":{\"deviceid\":\"e9f920\",\"tenant\":\"admin\",\"timestamp\":1598546364573},\"attrs\":{\"velocidade\":70}}"}
flowbroker_1                  | <16:39:24 27/08/2020> -- |dev cache -- DEBUG: retriving data related to admin:e9f920 from cache
flowbroker_1                  | <16:39:24 27/08/2020> -- DEBUG: [ingestor] got new device event: { metadata: { tenant: 'admin', timestamp: 1598546364573 },
flowbroker_1                  |   event: 'publish',
flowbroker_1                  |   data:
flowbroker_1                  |    { attrs:
flowbroker_1                  |       { velocidade: 70,
flowbroker_1                  |         carro: 'indefinido',
flowbroker_1                  |         device_timeout: '10000',
flowbroker_1                  |         protocol: 'mqtt' },
flowbroker_1                  |      id: 'e9f920' } }
flowbroker_1                  | <16:39:24 27/08/2020> -- DEBUG: [executor] will handle node ce40a438.e567c8:switch
flowbroker_1                  | <16:39:24 27/08/2020> -- |switch -- DEBUG: Executing switch node...
flowbroker_1                  | <16:39:24 27/08/2020> -- |switch -- DEBUG: ... switch node was successfully executed.
flowbroker_1                  | <16:39:24 27/08/2020> -- DEBUG: [executor] hop (switch) result: [{"payload":{"velocidade":70,"carro":"indefinido","device_timeout":"10000","protocol":"mqtt"}}]
flowbroker_1                  | <16:39:24 27/08/2020> -- DEBUG: Queued event [object Object]
flowbroker_1                  | <16:39:24 27/08/2020> -- DEBUG: [executor] will handle node 5d51ecfa.baa4e4:template
flowbroker_1                  | <16:39:24 27/08/2020> -- |template -- DEBUG: ... template node was successfully executed.
flowbroker_1                  | <16:39:24 27/08/2020> -- DEBUG: [executor] hop (template) result: [{"payload":{"velocidade":70,"carro":"indefinido","device_timeout":"10000","protocol":"mqtt"},"reqBody":"{\"payload\": \"velocidade muito alta: 70 km/h!\"}"}]
flowbroker_1                  | <16:39:24 27/08/2020> -- DEBUG: [executor] will handle node 784cd62a.09b088:http
flowbroker_1                  | <16:39:24 27/08/2020> -- |http -- DEBUG: Executing http node...
flowbroker_1                  | <16:39:24 27/08/2020> -- |http -- DEBUG: HTTP request about to be sent: Url {
flowbroker_1                  |   protocol: 'http:',
flowbroker_1                  |   slashes: true,
flowbroker_1                  |   auth: null,
flowbroker_1                  |   host: 'ptsv2.com',
flowbroker_1                  |   port: null,
flowbroker_1                  |   hostname: 'ptsv2.com',
flowbroker_1                  |   hash: null,
flowbroker_1                  |   search: null,
flowbroker_1                  |   query: null,
flowbroker_1                  |   pathname: '/t/9fbhu-1543424/post',
flowbroker_1                  |   path: '/t/9fbhu-1543424/post',
flowbroker_1                  |   href: 'http://ptsv2.com/t/9fbhu-1543424/post',
flowbroker_1                  |   method: 'POST',
flowbroker_1                  |   headers: {} }
flowbroker_1                  | <16:39:25 27/08/2020> -- |http -- DEBUG: ... http node was successfully executed.
flowbroker_1                  | <16:39:25 27/08/2020> -- DEBUG: [executor] hop (http) result: [{"payload":{"velocidade":70,"carro":"indefinido","device_timeout":"10000","protocol":"mqtt"},"reqBody":"{\"payload\": \"velocidade muito alta: 70 km/h!\"}","responseGet":{"statusCode":200,"headers":{"content-type":"text/plain; charset=utf-8","vary":"Accept-Encoding","access-control-allow-origin":"*","x-cloud-trace-context":"93dec034b3d9410c48f491bb80618d7a","date":"Thu, 27 Aug 2020 16:39:25 GMT","server":"Google Frontend","content-length":"54","connection":"close"},"responseUrl":"http://ptsv2.com/t/9fbhu-1543424/post","payload":"Thank you for this dump. I hope you have a lovely day!"}}]
```

![image](https://user-images.githubusercontent.com/37310063/91470985-cc99d600-e86b-11ea-9e0d-5ac926a3b2db.png)

- k8s

```
<17:28:57 27/08/2020> -- DEBUG: Got a message in topic 1adc623b-30ec-47ed-a380-f28bf8d21871.
<17:28:57 27/08/2020> -- DEBUG: There are 1 callbacks registered.
<17:28:57 27/08/2020> -- |messenger -- DEBUG: Received message: { value: <Buffer 7b 22 6d 65 74 61 64 61 74 61 22 3a 7b 22 64 65 76 69 63 65 69 64 22 3a 22 66 65 66 34 30 66 22 2c 22 74 65 6e 61 6e 74 22 3a 22 61 64 6d 69 6e 22 2c ... >,
  size: 103,
  key: null,
  topic: '1adc623b-30ec-47ed-a380-f28bf8d21871',
  offset: 5,
  partition: 0,
  timestamp: 1598549336898 }
<17:28:57 27/08/2020> -- |messenger -- DEBUG: Emitting new event message for subject device-data@admin
<17:28:57 27/08/2020> -- DEBUG: { '2e5735ff-7cd8-4eb0-a069-39a12914b300': [Function] }
<17:28:57 27/08/2020> -- DEBUG: Mapping device fef40f ===into===> event_queue3
<17:28:57 27/08/2020> -- DEBUG: Pre-processing event {"source":"device","message":"{\"metadata\":{\"deviceid\":\"fef40f\",\"tenant\":\"admin\",\"timestamp\":1598549336898},\"attrs\":{\"velocidade\":75}}"}
<17:28:57 27/08/2020> -- |dev cache -- DEBUG: retriving data related to admin:fef40f from cache
<17:28:57 27/08/2020> -- DEBUG: [ingestor] got new device event: { metadata: { tenant: 'admin', timestamp: 1598549336898 },
  event: 'publish',
  data:
   { attrs:
      { velocidade: 75,
        protocol: 'mqtt',
        device_timeout: '10000',
        carro: 'indefinido' },
     id: 'fef40f' } }
<17:28:57 27/08/2020> -- DEBUG: Queued event [object Object]
<17:28:57 27/08/2020> -- DEBUG: [executor] will handle node ce40a438.e567c8:switch
<17:28:57 27/08/2020> -- |switch -- DEBUG: Executing switch node...
<17:28:57 27/08/2020> -- |switch -- DEBUG: ... switch node was successfully executed.
<17:28:57 27/08/2020> -- DEBUG: [executor] hop (switch) result: [{"payload":{"velocidade":75,"protocol":"mqtt","device_timeout":"10000","carro":"indefinido"}}]
<17:28:57 27/08/2020> -- DEBUG: [executor] will handle node 5d51ecfa.baa4e4:template
<17:28:57 27/08/2020> -- |template -- DEBUG: ... template node was successfully executed.
<17:28:57 27/08/2020> -- DEBUG: [executor] hop (template) result: [{"payload":{"velocidade":75,"protocol":"mqtt","device_timeout":"10000","carro":"indefinido"},"reqBody":"{\"payload\": \"velocidade muito alta: 75 km/h!\"}"}]
<17:28:57 27/08/2020> -- DEBUG: [executor] will handle node 784cd62a.09b088:http
<17:28:57 27/08/2020> -- |http -- DEBUG: Executing http node...
<17:28:57 27/08/2020> -- |http -- DEBUG: HTTP request about to be sent: Url {
  protocol: 'http:',
  slashes: true,
  auth: null,
  host: 'ptsv2.com',
  port: null,
  hostname: 'ptsv2.com',
  hash: null,
  search: null,
  query: null,
  pathname: '/t/9fbhu-1543424/post',
  path: '/t/9fbhu-1543424/post',
  href: 'http://ptsv2.com/t/9fbhu-1543424/post',
  method: 'POST',
  headers: {} }
<17:28:57 27/08/2020> -- |http -- DEBUG: ... http node was successfully executed.
<17:28:57 27/08/2020> -- DEBUG: [executor] hop (http) result: [{"payload":{"velocidade":75,"protocol":"mqtt","device_timeout":"10000","carro":"indefinido"},"reqBody":"{\"payload\": \"velocidade muito alta: 75 km/h!\"}","responseGet":{"statusCode":200,"headers":{"content-type":"text/plain; charset=utf-8","vary":"Accept-Encoding","access-control-allow-origin":"*","x-cloud-trace-context":"e6756ce9f428182bd795c30de5b71f1e","date":"Thu, 27 Aug 2020 17:28:57 GMT","server":"Google Frontend","content-length":"54","connection":"close"},"responseUrl":"http://ptsv2.com/t/9fbhu-1543424/post","payload":"Thank you for this dump. I hope you have a lovely day!"}}]

```

![image](https://user-images.githubusercontent.com/37310063/91475343-d292b580-e871-11ea-97e8-fcee382a5907.png)
